File: auction-app-backend/src/main/java/com/internship/auctionapp/requests/UpdateUserDataRequest.java
package com.internship.auctionapp.requests;

import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
public class UpdateUserDataRequest {
    private UpdateUserRequest updateUserRequest;

    private CreateCreditCardRequest updateCreditCardRequest;

    private CreateAddressRequest updateAddressRequest;
}

File: auction-app-ui/src/requestModels/update/updateUserDataRequest.ts
import { CreateAddressRequest } from '../create/createAddressRequest';
import { CreateCreditCardRequest } from '../create/createCreditCardRequest';
import { UpdateUserRequest } from './updateUserRequest';

export type UpdateUserDataRequest = {
  updateUserRequest: UpdateUserRequest;
  updateCreditCardRequest: CreateCreditCardRequest;
  updateAddressRequest: CreateAddressRequest
};
